provision.sh

```diff
@@ -10,6 +16,19 @@ package() {
     sudo apt-get -y install "$@"
 }
 
+git_clone() {
+    # Clone git repo into directory, clearing directory if it exists
+    # Usage: git_clone URL DIR [GIT_CLONE OPTIONS...]
+
+    local url="$1"
+    shift
+    local dir="$1"
```

Comment:
> Because you cannot determine the name of the default destination directory from just the git URL

Isn't it just the last part of the path without the trailing `.git`?